keyboards/bastardkb/dilemma/4x6_4/4x6_4.c

```diff
@@ -57,3 +56,58 @@ bool encoder_update_kb(uint8_t index, bool clockwise) {
     return true;
 }
 #endif // ENCODER_ENABLE
+
+#ifdef RGB_MATRIX_ENABLE
+HSV _get_hsv_for_layer_index(uint8_t layer) {
+    switch (layer) {
+        case 1:
+            return (HSV){HSV_BLUE};
+        case 2:
+            return (HSV){HSV_AZURE};
+        case 3:
+            return (HSV){HSV_ORANGE};
+        case 4:
+            return (HSV){HSV_GREEN};
+        case 5:
+            return (HSV){HSV_TEAL};
+        case 6:
+            return (HSV){HSV_PURPLE};
+        case 7:
+        default:
+            return (HSV){HSV_RED};
+    };
+}
+
+// Layer state indicator
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+    if (host_keyboard_led_state().caps_lock) {
+        for (int i = led_min; i <= led_max; i++) {
+            if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
+                // set modifier-flagged LEDs to a pure red 76 steps brighter than the current brightness, clamped to 255 (ie. uint8_t max value).
+                rgb_matrix_set_color(i, MIN(rgb_matrix_get_val() + 76, 255), 0, 0);
+            }
+        }
+    }
+
+    uint8_t layer = get_highest_layer(layer_state);
+    if (layer > 0) {
+        HSV hsv = _get_hsv_for_layer_index(layer);
+
+        if (hsv.v > rgb_matrix_get_val()) {
+            // Set brightness slightly (22 steps) brighter than current brightness, clamped to 255 (ie. uint8_t max value)
+            hsv.v = MIN(rgb_matrix_get_val() + 22, 255);
```

Comment:
Should we simply set the brightness to the user's default value here? If the goal is to raise the brightness to at least a specific value, `rgb_matrix_get_val() + 22` seems a little arbitrary, may produce inconsistent results, and potentially override the user's preference.

Maybe we should simply set the value to `rgb_matrix_get_val()` unconditionally here instead:

```
HSV hsv = _het_hsv_for_layer_index(layer);
hsv.v = rgb_matrix_get_val();
```

In this case we don't need to clamp because we know the value is already in a valid range.

Reply:
Increasing the brightness slightly both makes the indicator more obvious, and allows it to work at a low brightness even if the brightness is set all the way down to disable animations. Disabling the indicators is still possible by turning off RGB entirely.
I find this behavior desirable. I suggest we continue this discussion outside of code review.

Reply:
Continuing here after having my flawed understanding of reviews challenged:

22 *is* rather arbitrary. The default step for the increase/decrease brightness keys seems to be 16, but `RGB_MATRIX_VAL_STEP` is defined where we can't easily include it, and ifndef define-ing it ourself seems contrary to the point of using a configurable default(?).

I'd like to keep the functionality that increasing the brightness a step above the user's preference brings. Do you have a suggestion of how to achieve this in a cleaner way?

Reply:
FYI, for traceability purposes only, my 2c shared on Discord:

> So here’s why I would find this behavior undesirable as a user: not only does this override the user’s preference (and forces a higher value than the user’s preferred max value), but it also prevents the user from reducing the brightness down to 0 (and essentially turning off their RGB via the brightness keycodes) because it ignores that value and turns the RGB back on.
> If we make an analogy to sound volume: imagine setting your max volume to 50%, but some software decides to ignore that and blast its sound at 75%. I doubt many people would like that. Similarly, imagine turning the volume down to 0%, and have it overridden by that software which is now making sound when you wanted it to be silent.
Visual effects, in particular potentially rapid changing ones like layers, can be very disruptive to some users, and an accessibility issue in some case. That’s why I think we should just respect whatever value the user has set, instead of overriding it with a higher one.
> The argument would be different at the keymap level, but this code is currently running at the keyboard level, and IMO needs to be less opinionated.

> There's 2 ways to turn off the RGB: enabling/disabling the feature, and turning the brightness down. Based on usage, users can use one or the other. It's not because it not a "me" usecase that it's not a valid use case (the same way people turn the volume down instead of muting it). And either way, even without the on/off argument, this is still forcing a higher brightness setting than what the user set, which is IMO forceful. Say you change the brightness to be not too strong in a dark room, but the keyboard ignores that and light up the room because it ignores the user's pref.

That being said, the consensus following that discussion is to keep the current behavior, mostly to simplify upstreaming and avoid any pushback from QMK's maintainers (which is expected to be limited to none in the current state since they already approved it for the Dilemma v2, and where even initially suggesting to move this code to the keyboard level).

Optionally, defines to parametrize the brightness increase value would be a good compromise, if not too much work.